### PR TITLE
Chore - Pre Commit

### PR DESCRIPTION
**Pull request by cusma**
# Intro

Hi all! 👋

This is Cosimo! MSc in Electrical Engineering at Politecnico di Torino. ⚡️

It is my first contribution on the repository.

# Scope

This PR is not touching any "core" electrical engineering feature, it is just improving the overall quality of the codebase and of the open source project.

The PR adds [pre-commit](https://pre-commit.com/) hooks to the repository:

> Git hook scripts are useful for identifying simple issues before submission to code review. We run our hooks on every commit to automatically point out issues in code such as missing semicolons, trailing whitespace, and debug statements. By pointing these issues out before code review, this allows a code reviewer to focus on the architecture of a change while not wasting time with trivial style nitpicks.

To run pre-commit:

1. Follow the [install](https://pre-commit.com/#install) guide
2. Execute `pre-commit run -a` in the folder containing the `.pre-commit-config.yaml` file

# Git hooks 

I've added the following git hooks to the `.pre-commit-config.yaml`:

- Checkov: Static code analysis for security and compliance (especially useful for GH actions)
- Pip Audit: Audits Python environments and dependency trees for known vulnerabilities
- Generic: 
  - Ensures that a file is either empty, or ends with one newline
  - Trims trailing whitespace
- YAML: Checks `.yaml` files for parseable syntax (again, in this repo would target mostly GH actions and pre-commit config)
- TOML: Checks `.toml` files for parseable syntax
- YAPF: Python code formatter

# Manual changes

- Added read permission to GH actions (best practice, suggested by Checkov)

# Automatic changes

⚠️ All the rest of PR `diff` is autogenerated according code formatting rules.

### CODE REVIEW

**Comment by cusma** [comment]
@mhinkkan I just saw the addition to the contribution guidelines in #149.

Would that be useful to include in the `pre-commit` the following validation (and discarding `black` and `ruff` proposed):

> We follow the `Style Guide for Python Code PEP8 <https://peps.python.org/pep-0008/>`_. Furthermore, we use the `YAPF Python code formatter <https://github.com/google/yapf>`_, configured based on the PEP8 conventions in the ``pyproject.toml`` file. Public methods and classes should have proper `docstrings <https://peps.python.org/pep-0257/>`_, formatted according to the `NumPy docstring guide <https://numpydoc.readthedocs.io/en/latest/format.html>`_. 

Or you want me to close this PR completely?

**Comment by mhinkkan** [comment]
@cusma, thank you for the PR! Your suggestion above sounds good, please consider modifying the PR accordingly.

File: examples/drive/flux_vector/plot_flux_vector_pmsm_2kw.py [hunk 1]
```diff
@@ -21,28 +26,27 @@
...
 # %%
 # Set the speed reference and the external load torque.
 
 # Speed reference (electrical rad/s)
-ctrl.ref.w_m = lambda t: (t > .2)*2*base.w
+ctrl.ref.w_m = lambda t: (t > 0.2) * 2 * base.w
 
 # Load torque step
-mdl.mechanics.tau_L = lambda t: (t > .8)*nom.tau*.7
+mdl.mechanics.tau_L = lambda t: (t > 0.8) * nom.tau * 0.7
```
Comment (mhinkkan) [inline examples/drive/flux_vector/plot_flux_vector_pmsm_2kw.py:49]: It seems that these modified files do not follow the yapf style defined in the pyproject.toml file for some reason.

Comment (cusma) [reply examples/drive/flux_vector/plot_flux_vector_pmsm_2kw.py]: Is this related to the `no_spaces_around_selected_binary_operators = "*,/"` rule? If so, could you check the `diff` of commit https://github.com/Aalto-Electric-Drives/motulator/pull/149/commits/b993b67010212de4643981729b6ea68fe91ea66b?

I've now included `additional_dependencies: [ toml ]` in the YAPF pre-commit hook.

**Comment by cusma** [comment]
> @cusma, thank you for the PR! Your suggestion above sounds good, please consider modifying the PR accordingly.

@mhinkkan I've removed `black` and `ruff` in favour of `yapf`, as per the contributing guidelines.

Also, I'm a bit hesitant in including the following NumPy Doc hooks:

```yaml
  # NumPy Doc
  - repo: https://github.com/numpy/numpydoc
    rev: v1.8.0
    hooks:
      - id: numpydoc-validation
        name: numpydoc-validation
        description: "This hook validates that docstrings in committed files adhere to numpydoc standards"
        entry: numpydoc lint
        require_serial: true
        language: python
        types: [ python ]
```

As after a little test it returns a lot of warnings about NumPy Doc validation.

**Comment by cusma** [comment]
@mhinkkan since there has been a bit of rework on this PR, if this looks mostly good I will close this PR in favour of a "clean" one, directly with YAPF (instead of the original Black & Ruff), so that the code changes are just related to the code formatting specified by the contributing guidelines.

**Comment by mhinkkan** [comment]
> since there has been a bit of rework on this PR, if this looks mostly good I will close this PR in favour of a "clean" one, directly with YAPF (instead of the original Black & Ruff), so that the code changes are just related to the code formatting specified by the contributing guidelines.

@cusma, this sounds good to me. Furthermore, I suggest to leave numpydoc-validation out (can appear as commented though).

**Comment by cusma** [comment]
Closed in favour of #160

**Pull request closed without merging**